core/src/main/java/org/elasticsearch/http/netty/NettyHttpRequest.java

```diff
@@ -69,6 +69,14 @@ public NettyHttpRequest(org.jboss.netty.handler.codec.http.HttpRequest request,
     @Override
     public Method method() {
         HttpMethod httpMethod = request.getMethod();
+
+        if (request.headers() != null) {
+            String methodOverride = request.headers().get("X-HTTP-Method-Override");
+            if (methodOverride != null) {
+                httpMethod = HttpMethod.valueOf(methodOverride);
+            }
+        }
+
```

Comment:
I want @jaymode to look at this one (see corresponding client usage in HttpClient.java).

Overall I really like this PR, and I like how it abstracts the implementation. But this hack might have implications... if its not ok, the only alternative I can think of right now is to use netty as the implementation?